pkgs/by-name/ml/mlibc/package.nix

```diff
@@ -0,0 +1,127 @@
+{
+  lib,
+  buildPackages,
+  stdenvNoLibc,
+  fetchFromGitea,
+  fetchFromGitHub,
+  fetchpatch,
+  ninja,
+  pkg-config,
+  linuxHeaders,
+}:
+
+let
+  freestnd-c-hdrs = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "osdev";
+    repo = "freestnd-c-hdrs";
+    rev = "4039f438fb1dc1064d8e98f70e1cf122f91b763b";
+    sha256 = "sha256-JMb1DS9+IDqGXdivMdy326kh+5iBniGc2RHihCfhLeA=";
+  };
+  freestnd-cxx-hdrs = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "osdev";
+    repo = "freestnd-cxx-hdrs";
+    rev = "85096df5361a4d7ef2ce46947e555ec248c2858e";
+    sha256 = "sha256-sQaIRgJgTLEci7kEJSsiGen/0jWSsyICqB5Ynk6XNu4=";
+  };
+  frigg = fetchFromGitHub {
+    owner = "managarm";
+    repo = "frigg";
+    rev = "ea0e9b5ea709b0c1b3e72ffc5de7019ca46ef427";
+    sha256 = "sha256-YmnyAdSmEJxGpvKy9pXXBWnNAY6XUcJIBBjU6VSW9Q0=";
+  };
+in
+stdenvNoLibc.mkDerivation (finalAttrs: {
+  pname = "mlibc";
+  version = "6.0.1-unstable-2025-07-03";
+
+  src = fetchFromGitHub {
+    owner = "managarm";
+    repo = "mlibc";
+    rev = "dee33115c03f2d80023651881e3814cb2d495c8d";
+    sha256 = "sha256-Qc6VTaG34A/2uh8We0INwK753MQQ0vUloleiFsfDZRE=";
+  };
+
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/managarm/mlibc/pull/1280.patch";
```

Comment:
Please read the guidance for patches in pkgs/README.md.  Unmerged patches should be vendored.